webapp/src/components/modals/create_event_form.tsx

```diff
@@ -54,14 +63,30 @@ export default function CreateEventForm(props: Props) {
         props.close();
     };
 
-    const handleError = (error: string) => {
-        setStoredError(error);
+    const handleError = (errorPayload: ErrorPayload) => {
+        const errorMessage = errorPayload.error.charAt(0).toUpperCase() + errorPayload.error.slice(1)
+        setStoredError(errorMessage);
         setSubmitting(false);
     };
 
-    const createEvent = async (payload: CreateEventPayload): Promise<{error?: string}> => {
-        alert(JSON.stringify(payload));
-        return {};
+    const createEvent = async (payload: CreateEventPayload): Promise<{ error?: string, data?: any }> => {
+        return new Promise((resolve, reject) => {
+            doFetchWithResponse('/plugins/com.mattermost.gcal/api/v1/events/create', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(payload),
+            }).
+                then((data) => {
+                    resolve(data);
+                }).
+                catch((response) => {
+                    if (response.status_code >= 400) {
+                        handleError(response.message);
+                    }
+                });
+        });
```

Comment:
`return new Promise` shouldn't usually be in userland code. Note that an `async` function implicitly returns a promise.

Ideally `doFetch` or `doFetchWithResponse` is not called directly by the component. It should instead be called in `actions.ts`, and export an action `createEvent`